core/spec/models/spree/shipping_method_spec.rb

```diff
@@ -231,20 +231,21 @@ def matching(categories)
     end
 
     context 'when the store has no shipping methods associated' do
+
       before { store.shipping_methods = [] }
 
       it 'returns all shipping methods' do
         expect(store.shipping_methods).to eq([])
-        expect(described_class.available_to_store(store)).to eq(subject)
+        expect(subject).to eq([first_shipping_method, second_shipping_method])
       end
     end
 
     context 'when the store has shipping methods associated' do
       before { create(:shipping_method) }
 
       it 'returns the associated records' do
-        expect(store.shipping_methods).to eq(subject)
-        expect(described_class.available_to_store(store)).to match_array(subject)
+        expect(store.shipping_methods).to eq([first_shipping_method, second_shipping_method])
+        expect(subject).to contain_exactly(first_shipping_method, second_shipping_method)
```

Comment:
If there's no order on these associations (I assume there isn't), then the spurious failures are likely due to the database returning the records in a different order.

This last assertion uses `contain_exactly`. Like `match_array`, it doesn't care about order. Maybe the other `.eq` assertions should be traded for one of those matchers?

Reply:
Thank you, good observation!